ginisdk/src/main/java/net/gini/android/DocumentTaskManager.java

```diff
@@ -93,6 +99,193 @@ public Document then(Task<JSONObject> task) throws Exception {
                 }
             };
 
+    /**
+     * Deletes a Gini partial document and all its parent composite documents.
+     * <br>
+     * Partial documents can be deleted only, if they don't belong to any composite documents and
+     * this method deletes the parents before deleting the partial document.
+     *
+     * @param documentId The id of an existing partial document
+     *
+     * @return A Task which will resolve to an empty string.
+     */
+    public Task<String> deletePartialDocumentAndParents(@NonNull final String documentId) {
+        return getDocument(documentId).onSuccessTask(new Continuation<Document, Task<Void>>() {
+            @Override
+            public Task<Void> then(Task<Document> documentTask) throws Exception {
+                final Document document = documentTask.getResult();
+                return deleteDocuments(document.getCompositeDocuments());
+            }
+        }, Task.BACKGROUND_EXECUTOR).onSuccessTask(new Continuation<Void, Task<Session>>() {
+            @Override
+            public Task<Session> then(final Task<Void> task) throws Exception {
+                return mSessionManager.getSession();
+            }
+        }, Task.BACKGROUND_EXECUTOR).onSuccessTask(new Continuation<Session, Task<String>>() {
+            @Override
+            public Task<String> then(final Task<Session> task) throws Exception {
+                final Session session = task.getResult();
+                return mApiCommunicator.deleteDocument(documentId, session);
+            }
+        });
+    }
+
+    /**
+     * Deletes a Gini document.
```

Comment:
I know that this deletes a non partial/composite document (old way). But maybe it is not obvious for those who use the library. Could you clarify a little bit?

Reply:
This performs a DELETE method request for the document id. Works with any document so it does what it says: `Deletes a Gini document.` 🙂. If you try to delete a partial document which is part of a composite document, than it will fail, but this exceeds the scope of this method and its javadoc. The specialized delete method `deletePartialDocumentAndParents()` handles that special case.